### PR TITLE
run: Get email from a user in the specified format.

### LINKED ISSUE #491: Validate usernames to be email adresses if `require_email_format_usernames` is set to True.
Validation can be done on the username entered by the user according to the value set in `require_email_format_usernames`.
See https://chat.zulip.org/#narrow/stream/206-zulip-terminal/topic/.5Bminor.20issue.5D.20Rephrasing.20the.20login.20error.20message/near/805059.

### PR DESCRIPTION
**Pull request by preetmishra**
This commit specifies the Email format according to the values set in
`require_email_format_usernames` and `email_auth_enabled`.
It also performs a validation check to see if the Email entered by a user
is valid.

Fixes #491.

### CODE REVIEW

**Comment by amanagr** [comment]
Thanks @preetmishra. This looks great. Next step would be to add test cases for the changes you have made in `tests/cli/test_run.py`.

**Comment by preetmishra** [comment]
@amanagr Thank you. I have added a commit that adds test for the changes. It is ready for review.

**Comment by neiljp** [comment]
@preetmishra I've been quite busy with other things recently, but am taking a look now.

File: zulipterminal/cli/run.py [hunk 1]
```diff
@@ -66,10 +67,33 @@ def parse_args(argv: List[str]) -> argparse.Namespace:
...
 
+def get_email(realm_url: str) -> str:
+    res_json = requests.get(url=realm_url + '/api/v1/server_settings').json()
+    require_email_format_usernames = res_json['require_email_format_usernames']
+    email_auth_enabled = res_json['email_auth_enabled']
+
+    if not require_email_format_usernames and email_auth_enabled:
+        email = input(in_color('blue', 'Email or Username: '))
+    elif not require_email_format_usernames:
+        email = input(in_color('blue', 'Username: '))
```
[the remark this answers is not in the document]
Comment (preetmishra) [reply zulipterminal/cli/run.py]: Yes, that'd be a lot cleaner. I'll change that.

File: zulipterminal/cli/run.py [hunk 2]
```diff
@@ -66,10 +67,33 @@ def parse_args(argv: List[str]) -> argparse.Namespace:
...
+    else:
+        # Validate Email as 'require_email_format_usernames' is True
+        while True:
+            email = input(in_color('blue', 'Email: '))
+            # RegEx: Checks whether Email address is of format '*@*.*'
+            regex_pattern = r'[^@]+@[^@]+\.[^@]+'
```
[the remark this answers is not in the document]
Comment (preetmishra) [reply zulipterminal/cli/run.py]: No, I wrote it. But you're right It'll end up failing for those two scenarios. I'll add a `TODO` there for now.

**Comment by preetmishra** [comment]
I agree. I haven't written much test code before. I am willing to put in the effort to write better tests.

**Comment by preetmishra** [comment]
@neiljp I have resolved the conflict and changed `get_email` to `get_login_id` as the value returned by the function can either be an email address or a username so `get_login_id` makes more sense. I have also changed variables related to it. Further, I have made the condition block cleaner using a label variable and added TODO for the email validation. 
I have removed the other test commit (tests: run:) from this PR for now as suggested.
It is ready for review.

**Comment by neiljp** [comment]
@preetmishra Thanks for working on this :+1: I've merged this commit with a change to the commit message (it doesn't fix the client side email check, so I removed the fixes part too), and followed it with refactor and test commits on master which should give an idea of how to move forward with more error handling.

**Pull request closed without merging**